component/LayoutContainer.tsx

```diff
@@ -0,0 +1,19 @@
+import React, { ReactNode } from 'react'
+
+type LayoutContainerProps = {
+  children: ReactNode;
+  backgroundImage?: string;
+}
+const LayoutContainer = ({ children, ...props }: LayoutContainerProps) => {
+  return (
+    <div className='flex flex-col items-center h-screen w-full bg-white md:max-w-M lg:max-w-M xl:max-w-M overflow-hidden p-4' style={props.backgroundImage && {
+      backgroundImage: `url('${props.backgroundImage}')`,
+      backgroundSize: 'cover',
+      width: '100%',
+    }}>
+      {children}
+    </div>
+  )
+}
+
+export default LayoutContainer
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `LayoutContainer` component seems to be well implemented. However, there's a minor issue with the way you're applying the background image style. If `backgroundImage` prop is not provided, an empty style object will be applied which is unnecessary. You can conditionally apply the style only when `backgroundImage` prop is present.

```diff
- <div className='flex flex-col items-center h-screen w-full bg-white md:max-w-M lg:max-w-M xl:max-w-M overflow-hidden p-4' style={props.backgroundImage && {
-   backgroundImage: `url('${props.backgroundImage}')`,
-   backgroundSize: 'cover',
-   width: '100%',
- }}>
+ <div className={`flex flex-col items-center h-screen w-full bg-white md:max-w-M lg:max-w-M xl:max-w-M overflow-hidden p-4 ${props.backgroundImage ? "bg-cover" : ""}`} 
+   style={props.backgroundImage ? { backgroundImage: `url('${props.backgroundImage}')`, width: '100%' } : {}}
+ >
    {children}
  </div>
```
This change ensures that the style object is only applied when `backgroundImage` prop is present, improving readability and performance slightly by avoiding unnecessary object creation and assignment.


<!-- This is an auto-generated comment by OSS CodeRabbit -->